src/components/interpolatedColorLegend.ts

```diff
@@ -0,0 +1,249 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Component {
+  export class InterpolatedColorLegend extends AbstractComponent {
+    private _measurer: SVGTypewriter.Measurers.Measurer;
+    private _wrapper: SVGTypewriter.Wrappers.Wrapper;
+    private _writer: SVGTypewriter.Writers.Writer;
+    private _scale: Scale.InterpolatedColor;
+    private _orientation: String ;
+    private _padding = 5;
+    private _formatter: Formatter;
+
+    private _swatchContainer: D3.Selection;
+    private _lowerLabel: D3.Selection;
+    private _upperLabel: D3.Selection;
+
+    /**
+     * The css class applied to the legend labels.
+     */
+    public static LEGEND_LABEL_CLASS = "legend-label";
+
+    /**
+     * Creates an InterpolatedColorLegend.
+     *
+     * The InterpolatedColorLegend consists of a sequence of swatches, showing the
+     * associated Scale.InterpolatedColor sampled at various points. Two labels
+     * show the maximum and minimum values of the Scale.InterpolatedColor.
+     *
+     * @constructor
+     * @param {Scale.InterpolatedColor} interpolatedColorScale
+     * @param {string} orientation (horizontal/left/right).
+     * @param {Formatter} The labels are formatted using this function.
+     */
+    constructor(interpolatedColorScale: Scale.InterpolatedColor, orientation = "horizontal", formatter = Formatters.general()) {
+      super();
+      if (interpolatedColorScale == null ) {
+        throw new Error("InterpolatedColorLegend requires a interpolatedColorScale");
+      }
+      this._scale = interpolatedColorScale;
+      this._formatter = formatter;
+      this._orientation = InterpolatedColorLegend.validateOrientation(orientation);
+
+      this._fixedWidthFlag = true;
+      this._fixedHeightFlag = true;
+    }
+
+    /**
+     * Gets the current formatter on the InterpolatedColorLegend.
+     *
+     * @returns {Formatter} The current Formatter.
+     */
+    public formatter(): Formatter;
+    /**
+     * Sets the current formatter on the InterpolatedColorLegend.
+     *
+     * @param {Formatter} formatter If provided, data will be passed though `formatter(data)`.
+     * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
+     */
+    public formatter(formatter: Formatter): InterpolatedColorLegend;
+    public formatter(formatter?: Formatter): any {
+      if (formatter === undefined) {
+        return this._formatter;
+      }
+      this._formatter = formatter;
+      this._invalidateLayout();
+      return this;
+    }
+
+    private static validateOrientation(orientation: string) {
```

Comment:
Mainly semantical, but I like the term ensureOrientation rather than validateOrientation since I feel that validateOrientation doesn't sound like a term that would return me a corrected orientation if it passes the validation check.  I feel that ensure essentially tells me that the return value will be a guaranteed good orientation or an error will be thrown.